### ISSUE TITLE
HTTP status

### ISSUE THREAD

**Issue by cylkee** [symphonycms/maintenance_mode#19]
Hi,

For anonymous visitors, should not Maintenance Mode throw an HTTP status of [503 Service Unavailable][1]?  I'm thinking for search engines in particular so they know not to try to reindex a maintenance page for every URL in one's site as currently it appears to send 200 OK.

Thanks!

  [1]: http://tools.ietf.org/html/rfc7231#section-6.6.4

**Comment by michael-e** [comment]
Yes, it should, probably. But for perfection you might also want a (customizable) `Retry-After` header.

You can do that with the [HTTP Response Header Mappings](https://github.com/michael-e/http_response_header_mappings) extension. In the config file you could set:

```php
'503' => 'HTTP/1.1 503 Service Unavailable',
'ra1h' => 'Retry-After: 3600',
'ra1d' => 'Retry-After: 86400',
'ra1w' => 'Retry-After: 604800',
```

then add the needed page types (e.g. `503` and `ra1d`) to your maintenance page.

**Comment by nitriques** [comment]
@michael-e I think this extension should do it.

**Comment by michael-e** [comment]
Yep.

### Solution 1

It might at least create some (documented) array in the config file. Instead of this:

```php
'maintenance_mode' => array(
	'enabled' => 'no',
),
```

it could be:

'maintenance_mode' => array(
	'enabled' => 'no',
	'status_header' => 'HTTP/1.1 503 Service Unavailable',
	'retry_after_header' => 'Retry-After: 3600',
),

If a header is not found in the config, it should not be sent.

### Solution 2

To keep things simple, we might add the 503 header only. If someone needs more advanced headers (like above), he can still use the HTTP Response Header Mappings extension. 

I prefer solution 2.

**Comment by nitriques** [comment]
> I prefer solution 2.

Me too. And the status_header can only be the numeric status code since [we already have the mapping to string](https://github.com/symphonycms/symphony-2/blob/master/symphony/lib/toolkit/class.page.php#L90-L104).

**Comment by michael-e** [comment]
Agreed.

**Comment by brendo** [comment]
This is a good idea! I'm thinking we can send `503` by default, and then read a `Retry` value from the configuration? Or should we just use a sane retry default and if someone wants to modify it then they use the Response Mapping extension?

**Comment by michael-e** [comment]
> I'm thinking we can send 503 by default, and then read a Retry value from the configuration?

I like this idea. As long as the config's `Retry-after` value is empty (which is the default), we just send a 503. If it's not empty, we send this header as well. No additional extensions needed.

**Comment by jensscherbl** [comment]
> Or should we just use a sane retry default and if someone wants to modify it then they use the Response Mapping extension?

Sane default, please.

**Comment by michael-e** [comment]
The right value might be 15 minutes (if you take your site down for an upgrade or similar) or several days (during development). I don't see a sane default.

**Comment by animaux** [comment]
Anyone still into this?

**Comment by nitriques** [comment]
@animaux I am still into it, but I have other things more urgent...

**Comment by cylkee** [comment]
I gave up on Maintenance Mode and I just modify .htaccess in cases where I need `503`.  I think Google _et al_ will mostly ignore `Retry-after`, since they do things on their own time-frame.  Debating over finer detail like that just delays what I think is quite an important update to this extension.

Thanks for self-assigning, @nitriques.

**Comment by animaux** [comment]
@nitriques thanks, it’s not urgent but a typical nice-to-have :)

**Comment by nitriques** [comment]
Indeed ;)
